packages/playground/src/explorer/components/node_details_cards/twin_details_card.vue

```diff
@@ -17,38 +18,58 @@ export default {
   props: {
     node: {
       type: Object as PropType<GridNode>,
-      required: true,
+      required: false,
+    },
+    farm: {
+      type: Object as PropType<Farm>,
+      required: false,
     },
   },
 
   setup(props) {
     const loading = ref<boolean>(false);
     const twinFields = ref<NodeDetailsCard[]>();
+    const twin = ref<Twin>();
 
-    const mount = () => {
+    onMounted(async () => {
       loading.value = true;
+
+      if (props.farm) {
+        const { data } = await getTwins({
+          twinId: props.farm.twinId,
+        });
+        twin.value = data[0];
+      } else if (props.node) {
+        twin.value = props.node.twin;
+      }
       twinFields.value = getNodeTwinDetailsCard();
       loading.value = false;
-    };
-
-    onMounted(mount);
+    });
 
     const copy = (address: string) => {
       navigator.clipboard.writeText(address);
       createCustomToast("Copied!", ToastType.success);
     };
 
     const getNodeTwinDetailsCard = (): NodeDetailsCard[] => {
+      let twinId, accountId, relay;
+      if (twin.value) {
```

Comment:
What is twin doesn't have value?

Reply:
handled by adding default values if twin is undefined, thanks